packages/cozy-client/src/associations/helpers.js

```diff
@@ -22,15 +22,12 @@ export const responseToRelationship = response =>
   })
 
 const attachRelationship = (doc, relationships) => {
-  if (
-    doc.relationships &&
-    isEqual(Object.keys(doc.relationships), Object.keys(relationships))
-  ) {
-    return doc
-  }
   return {
     ...doc,
-    relationships: relationships || {}
+    relationships: {
+      ...doc.relationships,
+      ...relationships
```

Comment:
When destructuring, please make sure that the object exists. A test for the case where `relationships` is undefined should catch the failure ;)